torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +72,95 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
+    quant_min: Optional[float] = None
+    quant_max: Optional[float] = None
+    scale_min: Optional[float] = None
+    scale_max: Optional[float] = None
+
+
 @dataclass
 class DTypeConfig:
     """
     Config for the set of supported input/output activation, weight, and bias data types for the
     patterns defined in :class:`~torch.ao.quantization.backend_config.BackendConfig`.
     """
-    input_dtype: Optional[torch.dtype] = None
-    output_dtype: Optional[torch.dtype] = None
-    weight_dtype: Optional[torch.dtype] = None
+    input_dtype: Optional[DTypeWithConstraints] = None
```

Comment:
maybe the `dtype` inside `DTypeWithConstraints` can be None?

Reply:
That would mean the user could specify quant_min/quant_max without specifying which dtype this range is for. I'm not sure if that makes sense, since these restrictions should be tied to particular dtypes